Solutions/GoogleWorkspaceReports/Data Connectors/GWorkspaceReportsAPISentinelConnector/__init__.py

```diff
@@ -158,20 +212,82 @@ def gen_chunks_to_object(data,chunksize=100):
     yield chunk
 
 def gen_chunks(data,log_type):
-    for chunk in gen_chunks_to_object(data, chunksize=2000):
-        body = json.dumps(chunk)
-        post_data(customer_id, shared_key,body,log_type)
+    #for chunk in gen_chunks_to_object(data, chunksize=2000):
+        #body = json.dumps(chunk)
+        #post_data(customer_id, shared_key,body,log_type)
+    chunks = [data[i:i+chunksize] for i in range(0, len(data), chunksize)]
+    logging.info("Entered into the chunks mode")
+    latest_timestamp = "";
+    i = 0
+    for chunk in chunks:
+        try:
+            i = i+1
+            logging.debug("Iteration chunk {}".format(i))
+            body = json.dumps(chunk)
+            logging.debug(body)
+            statuscode = post_data(customer_id, shared_key,body,log_type, len(chunk))
+            if (statuscode >= 200 and statuscode <= 299):
+                latest_timestamp = chunk[-1]["id"]["time"]
+                dt = datetime.strptime(latest_timestamp, '%Y-%m-%dT%H:%M:%S.%fZ')
+                dt += timedelta(milliseconds=1)
+                latest_timestamp = dt.strftime('%Y-%m-%dT%H:%M:%S.%f')
+                latest_timestamp = latest_timestamp[:-3] + 'Z'
+                logging.info("Chunk Timestamp {}".format(latest_timestamp))
+                #logging.info("Logs with {} activity was processed into Azure".format(log_type))
+                #logging.info("Chunk was processed{} events".format(chunk_count))
+            else:
+                logging.warn("There is an issue with Posting data to LA - Response code: {}".format(statuscode))
+        except Exception as err:
+            logging.error("Something wrong. Exception error text: {}".format(err))
+    return latest_timestamp
```

Comment:
Are we covering the scenario where only few chunks are successful and failure after that?

Reply:
yes handling this scenario